back/internal/usersvc/internal/userdb/qryuser.go

```diff
@@ -71,21 +77,20 @@ func (s *UserDB) upsertUser(ctx cx, sid string, x *pb.AddUserReq, y *pb.UserResp
 		x.Password,
 	)
 	if err != nil {
+		tx.Rollback()
 		return err
 	}
 
-	// Add role and user to user_role join table.
-	stmt, err := s.db.Prepare("INSERT into user_role (user_id, role_id) VALUES (?, ?)")
+	qry = "INSERT into user_role (user_id, role_id) VALUES "
+	vals, args := buildValsAndArgs(id.String(), x.RoleIds)
+	_, err = tx.ExecContext(ctx, qry+vals, args...)
 	if err != nil {
+		tx.Rollback()
```

Comment:
If this or other rollback calls return an error, please make a new issue to implement handling them. We can:
1. log
2. retry
3. do nothing different
4. check what other people do

Consider making a related issue to create a logging policy (or look around to see if we are doing any logging already).

Reply:
I've created the two issues you suggested:
- Create logging policy #123
- Handle errors from rollback calls #124